app/jsx/pages/BrowsePage.jsx

```diff
@@ -105,11 +106,15 @@ class AllCampuses extends React.Component {
 class AllJournals extends React.Component {
   constructor(props) {
     super(props)
-    this.state = {campusID: props.campusID,
+    this.state = {campusID: "",
                   isActive: props.isActive}
     this.changeCampus = this.changeCampus.bind(this)
   }
 
+  componentWillUnmount() {
```

Comment:
I noticed the input selector wasn't being reset when I clicked back to this page. Ideally it would be nice to preserve all state on this page, I wasn't able to get that working. For now, if you click back to this page, all journals display (and input selector should be set to "All Campuses").